src/hal/src/command/mod.rs

```diff
@@ -65,100 +64,131 @@ impl Level for Primary { }
 pub enum Secondary { }
 impl Level for Secondary { }
 
-/// Thread-safe finished command buffer for submission.
-pub struct Submit<B: Backend, C, S, L>(pub(crate) B::CommandBuffer, pub(crate) PhantomData<(C, S, L)>);
-impl<B: Backend, C, S, L> Submit<B, C, S, L> {
-    fn new(buffer: B::CommandBuffer) -> Self {
-        Submit(buffer, PhantomData)
-    }
+/// A property of a command buffer to be submitted to a queue.
+pub trait RawSubmittable<B: Backend> {
+    /// Borrow a raw command buffer from this submittable.
+    fn as_command_buffer(&self) -> &B::CommandBuffer;
 }
-unsafe impl<B: Backend, C, S, L> Send for Submit<B, C, S, L> {}
 
+/// A property of a command buffer to be submitted to a queue with specific capability.
+pub trait Submittable<B: Backend, K: Capability, L: Level>: RawSubmittable<B> {}
```

Comment:
In all other places `C` is used for `Capability`. Except implementation of this trait, where two capability type parameters are defined and using `K` perfectly justified.